Description: Allow overriding of AGexe (for cross build)
 build-aux/run-ag.sh hard codes some path for AGexe, which makes the
 cross build unable to find a working autogen.
 The relevant callers of run-ag.sh fail to pass the correct AGexe.
Author: Helmut Grohne <helmut@subdivi.de>
Origin: vendor
Bug-Debian: https://bugs.debian.org/941025
Forwarded: no
Reviewed-By: Andreas Metzler <ametzler@debian.org>
Last-Update: 2019-10-05

--- build-aux/run-ag.sh
+++ build-aux/run-ag.sh
@@ -25,7 +25,6 @@
 # any containing directory must be created. The target is created with a
 # very old time stamp.
 #
-AGexe=/u/bkorb/tools/ag/autogen-bld/agen5/.libs/autogen
 find_exe() {
   eval local exe=\${$1}
   test -x "$exe" && return 0
--- getdefs/Makefile.am
+++ getdefs/Makefile.am
@@ -32,7 +32,7 @@ nodist_getdefs_SOURCES = $(BUILT_SOURCES
 SUBDIRS         = test
 EXTRA_DIST      = opts.def $(gdsrcs)
 AG_ENV          =  top_builddir="$(top_builddir)" top_srcdir="$(top_srcdir)" \
-	VERBOSE="$(V)"
+	VERBOSE="$(V)" AGexe="$(AGexe)"
 RUN_AG          = $(AG_ENV) $(SHELL) "${top_srcdir}/build-aux/run-ag.sh"
 
 all : gen
--- xml2ag/Makefile.am
+++ xml2ag/Makefile.am
@@ -37,7 +37,7 @@ DISTCLEANFILES  = $(DOCFILES) $(nodist_x
 AM_CPPFLAGS     = @INCLIST@ $(LIBXML2_CFLAGS)
 AM_CFLAGS       = @WARN_CFLAGS@
 AG_ENV          =  top_builddir="$(top_builddir)" top_srcdir="$(top_srcdir)" \
-	VERBOSE="$(V)"
+	VERBOSE="$(V)" AGexe="$(AGexe)"
 RUN_AG       = $(AG_ENV) $(SHELL) "${top_srcdir}/build-aux/run-ag.sh"
 DOC_TIMEOUT     = -DLEVEL=section --timeout=`expr $(AG_TIMEOUT) '*' 3`
 
